### ISSUE TITLE
Shellcheck does not handle namerefs correctly

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#2254]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2034, SC2034
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
# namerefs
myfunc ()
{
    declare -n first=$1
    declare -n second=$2
    first="hi there"
    second="bye there"
}

declare eek
declare oof

myfunc eek oof
echo "eek=$eek, oof=$oof"

```

#### Here's what shellcheck currently says:
```
Line 7	SC2034: first appears unused. Verify use (or export if used externally).
Line 8	SC2034: second appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:
Nothing.  Assignment to a nameref causes the data to be available in the caller. Using a `declare -n` and NOT assigning to it should be warning, like: `Using a nameref, but not assigning to it. Consider using traditional pass-by-value instead.`
